cmd/ls/ls_test.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var tempDir string
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+	os.Exit(code)
+}
+
+func setup() {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		fatal(err)
+	}
+
+	tempDir = dir
+	createTempFiles(tempDir)
+}
+
+func teardown() {
+	os.RemoveAll(tempDir)
+}
+
+func createTempFiles(dir string) {
+	content := []byte("temporary file's content")
+	files := []string{"file with space", "f1.txt", "f2.pdf", "f3"}
+
+	for _, f := range files {
+		fileName := filepath.Join(dir, f)
+		if err := ioutil.WriteFile(fileName, content, 0666); err != nil {
+			fatal(err)
+		}
+	}
+}
+
+func TestListCurrentDir(t *testing.T) {
+	var buf bytes.Buffer
+	files, _ := ioutil.ReadDir(tempDir)
+	printFileList(files, &buf)
+	output := string(buf.Bytes())
+	expected := "-rw-rw-r--   24 f1.txt\n-rw-rw-r--   24 f2.pdf\n-rw-rw-r--   24 f3\n-rw-rw-r--   24 'file with space'\n"
+	if output != expected {
+		t.Errorf("result not expected [%v]", output)
+	}
+}
+
+func TestListCurrentDirAsAList(t *testing.T) {
+	var buf bytes.Buffer
+	files, _ := ioutil.ReadDir(tempDir)
+	printFileNames(files, &buf)
+	output := string(buf.Bytes())
+	expected := "f1.txt f2.pdf f3 'file with space' \n"
+	if output != expected {
+		t.Errorf("result not expected [%v]", output)
+	}
+}
+
+func TestHumanizeBytes(t *testing.T) {
```

Comment:
There is a good instance of Table Driven Tests on this repeated ifs. You could have a list of (val,result) and iterate on it checking if it works. It would be more declarative and easier to add new cases :-)